### DISCUSSION TITLE
StatusEffectAction causes obvious lag spike in multiplayer

### DISCUSSION THREAD

**Discussion by A2ur3** [FakeFishGames/Barotrauma#14916] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

```xml
<ScriptedEvent identifier="abandoned_one">
  <TagAction Criteria="crew" Tag="crew" IgnoreIncapacitatedCharacters="false" />

  <Label Name="eventloop" />
  <WaitAction Time="0.9" />
  <StatusEffectAction TargetTag="crew">
    <StatusEffect target="This" disabledeltaTime="true" multiplyafflictionsbymaxvitality="true">
      <Conditional IsInPlayerSub="true" />
      <Affliction identifier="abandonedone" strength="1.0" />
    </StatusEffect>
  </StatusEffectAction>
  <GoTo Name="eventloop" />
</ScriptedEvent>
```
I'm using this event to give afflictions to crew inside submarine
It worked fine before but in current multiplayer the event will cause obvious lag
"abandonedone" can be replaced with vanilla affliction like "organdamage"

### Reproduction steps

1. start a multiplayer campaign
2. trigger the event above
3. lag

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.6.17.0 (Unto the Breach Update)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report! I gave this a test, but unfortunately couldn't spot any issues. What do you mean by "lag" specifically? Latency between the client and server? FPS drops, freezes? Input lag? I wonder if it could be the custom "abandonedone" affliction contributes to this somehow - I tested this with the vanilla bleeding affliction?

https://github.com/user-attachments/assets/161d1679-369a-49fe-b109-0105692eb1f0

**Reply by A2ur3** [reply]
I have posted a video below

**Comment by A2ur3** [comment]
https://github.com/user-attachments/assets/92e56e40-d986-492a-9e81-ca3cd887207c

**Comment by A2ur3** [comment]
I'm testing with mod BT-EW-T.S.M EventSet ALPHA and I've narrowed the issue down to randomevents_campaign.xml

**Reply by A2ur3** [reply]
the issue can be reproduced by testing event mentioned above alongside mod which overhauls randomevents_campaign.xml like Barotraumatic

**Comment by Rokvach** [comment]
Fixed and tested the fix to be working correctly, fix coming in the next hotfix update. Closing discussion.

**Reply by A2ur3** [reply]
is it included in current unstable autumn-update-hotfix-2?
I am testing with the event and the issue seems still exist

New reproduction steps:
1. download and enable Barotraumatic
2. go to submarine editor and spawn a funbringer 3000
3. equip the funbringer 3000 and use it to fire banana peel
4. game freezes

**Reply by Regalis11** [reply]
It seems we actually messed this up, and merged the fix to an incorrect branch (the branch of the first hotfix)! So no, it seems the fix is currently not included in the unstable branch. I'm glad you spotted this before we pushed the hotfix live. Sorry for the inconvenience, we'll address this!

**Comment by A2ur3** [comment]
Fixed. Closing discussion.
